src/base/cache/associative_cache.hh

```diff
@@ -48,12 +60,14 @@ namespace gem5
 template <typename Entry>
 class AssociativeCache : public Named
 {
-    static_assert(std::is_base_of_v<CacheEntry, Entry>,
-                  "Entry should be derived from CacheEntry");
+  protected:
 
-    typedef replacement_policy::Base BaseReplacementPolicy;
+    static_assert(std::is_base_of_v<ReplaceableEntry, Entry>,
```

Comment:
Does this work if Entry is directly derived from ReplaceableEntry instead of CacheEntry?

Reply:
Yes, as long as it complies with the template interface… In fact the TaggedEntry inherits directly from ReplaceableEntry now!

Reply:
In fact I believe some of the CacheEntry methods might be devirtualised now. They were probably made virtual to accomodate for the TaggedEntry

Reply:
thanks, the changes LGTM.